src/mlpack/methods/ann/layer/isrlu_impl.hpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file methods/ann/layer/isrlu_impl.hpp
+ * @author Abhinav Anand
+ *
+ * Implementation of the ISRLU activation function as described by Jonathan T. Barron.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_ISRLU_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_ISRLU_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "isrlu.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+ISRLU<InputDataType, OutputDataType>::ISRLU(const double alpha) :
+    alpha(alpha),
+    deterministic(false)
+{}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void ISRLU<InputDataType, OutputDataType>::Forward(
+    const InputType& input, OutputType& output)
+{
+  output = arma::ones<OutputDataType>(arma::size(input));
+  for (size_t i = 0; i < input.n_elem; ++i)
+  {
+    output(i) = (input(i) >= 0) ? input(i) : input(i) *
+                (1 / std::sqrt(1 + alpha*input(i)*input(i)));
```

Comment:
```suggestion
                (1 / std::sqrt(1 + alpha * input(i) * input(i)));
```

Missing space between the operation, also you have to use ` input(i) % input(i)` instead of ` input(i) * input(i)` since you want element-wise multiplication, which is important if `batch-size > 1`.

Reply:
this   gives an  error

Reply:
Ahh, right because `input(i)` is a scalar, so all good.

Reply:
any more changes